geoSegment/geoSegment.js

```diff
@@ -254,17 +258,23 @@ async function processSessionSeriesItems(items, segments) {
       'oo:segment': [],
     };
 
-    // Add oo:segment if applicable
-    for (const segment of segments) {
-      const segmentRadiusInMeters = segment.radius * 1000;
-      const distanceBetweenGeos = geolib.getDistance(
-        { latitude: sessionSeriesPhysicalLocationGeo.latitude, longitude: sessionSeriesPhysicalLocationGeo.longitude },
-        { latitude: segment.latitude, longitude: segment.longitude },
-      );
-      if (distanceBetweenGeos <= segmentRadiusInMeters) {
-        sessionSeriesData['oo:segment'].push(segment.identifier);
-      }
+    if (!sessionSeriesPhysicalLocationGeo) {
+      sessionSeriesData["oo:segment"] = segments.map(segment => segment.identifier);
+    } else {
```

Comment:
We probably don't need to bother with having `sessionSeriesPhysicalLocationGeo` as null for online-only locationless sessions.

This whole section could be simplified if we did something like (pseudo code):

```js
const sessionSeriesSegments = (() => {
  // If item is online-only and has no location, return all segment identifiers
  // If item has no location and is not online-only, return no segment identifiers
  // Otherwise, get segments using distance, as presently in the code
})();

// If there are no SessionSeries segments, drop this SessionSeries, as presently in the code

/** @type {SessionSeriesData} */
const sessionSeriesData = {
  ...item.data,
  'oo:segment': sessionSeriesSegments,
};
```

I think this would be more explicit than using `sessionSeriesPhysicalLocationGeo === null` as a stand-in for "This is an online-only location-less SessionSeries"